pkg/kubectl/categories/categories.go

```diff
@@ -86,14 +99,18 @@ func (e discoveryCategoryExpander) Expand(category string) ([]schema.GroupResour
 	return ret, ok
 }
 
+// discoveryFilteredExpander provides a reverse way comparing to discoveryCategoryExpander.
+// It expends the given CategoryExpander (delegate) which can be seen as a filter, to groupResouce list first (Like a 'target list'),
+// and then request the APIResourceList from server, convert it to groupResouce list based on the server (Like a 'actual list'),
+// lastly compare the two lists, keeping the same elements in result. It performs like a 'Filter'
 type discoveryFilteredExpander struct {
```

Comment:
better try to be shorter. A four line comment like that is hard to maintain and read. What does it do in one sentence?